### PR TITLE
Test serving images via Imgix for money section

### PR DESCRIPTION
**Pull request by phamann**
We are now in a position to do a production test using the Imgix image resizing service for our responsive images. To start with we are going to run this on a small proportion of traffic for which we have chosen the Money section.

The code is _quite hacky™_ as it's just a test.

/cc @gklopper @rich-nguyen @robertberry

### CODE REVIEW

File: common/app/views/fragments/meta/bylineImage.scala.html [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+@(tags: Tags)(implicit request: RequestHeader)
 
 @import views.support.Profile
+@import conf.Switches.{ImgixSwitch}
 
 @tags.contributors.headOption.map { profile =>
     @profile.contributorLargeImagePath.map{ src =>
         <div class="byline-img">
-            <img class="byline-img__img" src="@ImgSrc(src, Item300)" alt="@profile.name" />
+            <img class="byline-img__img" src="@ImgSrc(src, Item300, Seq("/sport", "/football").exists(request.path.startsWith) && ImgixSwitch.isSwitchedOn)" alt="@profile.name" />
```
Comment (phamann) [inline common/app/views/fragments/meta/bylineImage.scala.html]: This shouldn't be here. Removing now.

File: common/app/views/fragments/img.scala.html [hunk 2]
```diff
@@ -52,12 +53,14 @@
...
-        sizes="@widths.sizes"
-        class="maxed responsive-img"
-        itemprop="contentURL representativeOfPage"
-        alt="@ImgSrc.getFallbackAsset(picture).flatMap(_.altText).getOrElse("")" />
+    srcset="@ImgSrc.srcset(picture, widths)"
+    sizes="@widths.sizes"
+    class="maxed responsive-img"
+    itemprop="contentURL representativeOfPage"
+    alt="@ImgSrc.getFallbackAsset(picture).flatMap(_.altText).getOrElse("")" />
+
```
Comment (rich-nguyen) [inline common/app/views/fragments/img.scala.html]: just whitespace?

File: common/app/views/fragments/meta/bylineImage.scala.html [hunk 3]
```diff
@@ -1,6 +1,7 @@
-@(tags: Tags)
+@(tags: Tags)(implicit request: RequestHeader)
 
 @import views.support.Profile
+@import conf.Switches.{ImgixSwitch}
```
Comment (rich-nguyen) [inline common/app/views/fragments/meta/bylineImage.scala.html]: uneeded

File: common/app/views/support/Profile.scala [hunk 4]
```diff
@@ -94,8 +99,14 @@ object ImgSrc extends Logging {
...
         .map( pathPrefix =>
-        s"$imageHost/$pathPrefix${imageType.resizeString}${uri.getPath}"
+          if(imgix) {
+            s"http://$pathPrefix-guim.imgix.net${uri.getPath}${imageType.imgixResizeString}"
+          } else {
+            s"$imageHost/$pathPrefix${imageType.resizeString}${uri.getPath}"
+          }
         ).getOrElse(url)
+
+
```
Comment (rich-nguyen) [inline common/app/views/support/Profile.scala]: whitespace

File: common/app/views/support/Profile.scala [hunk 5]
```diff
@@ -110,20 +121,40 @@ object ImgSrc extends Logging {
...
+        } mkString ", "
+      } else {
+        normalSrcset(imageContainer, widths)
+      }
+  }
+
+  def normalSrcset(imageContainer: ImageContainer, widths: WidthsByBreakpoint): String = {
     widths.profiles.map { profile =>
-      s"${findSrc(imageContainer, profile).get} ${profile.width.get}w"
+        s"${findSrc(imageContainer, profile).get} ${profile.width.get}w"
```
Comment (rich-nguyen) [inline common/app/views/support/Profile.scala]: whitespace

File: common/app/views/support/Profile.scala [hunk 6]
```diff
@@ -110,20 +121,40 @@ object ImgSrc extends Logging {
     }
   }
 
-  def srcset(imageContainer: ImageContainer, widths: WidthsByBreakpoint): String = {
+  def findLargestSrc(imageContainer: ImageContainer, profile: Profile)(implicit request: RequestHeader): Option[String] = {
```
Comment (rich-nguyen) [inline common/app/views/support/Profile.scala]: `private`

File: common/app/views/support/Profile.scala [hunk 7]
```diff
@@ -132,7 +163,7 @@ object ImgSrc extends Logging {
 }
 
 object SeoThumbnail {
-  def apply(metadata: MetaData): Option[String] = metadata match {
+  def apply(metadata: MetaData)(implicit request: RequestHeader): Option[String] = metadata match {
```
Comment (rich-nguyen) [inline common/app/views/support/Profile.scala]: is this param needed?

**Comment by rich-nguyen** [comment]
:+1:

File: common/app/implicits/Requests.scala [hunk 8]
```diff
@@ -28,6 +32,8 @@ trait Requests {
 
     lazy val isHealthcheck: Boolean = r.headers.keys.exists(_ equalsIgnoreCase  "X-Gu-Management-Healthcheck")
 
+    lazy val isInImgixTest: Boolean = imgixTestSections.exists(r.path.startsWith) && Switches.ImgixSwitch.isSwitchedOn
```
Comment (robertberry) [inline common/app/implicits/Requests.scala]: Should this be a `val` rather than a `def`? (thinking about the switch here)

**Comment by robertberry** [comment]
:+1:

File: common/app/implicits/Requests.scala (hunk 8 above)
Comment (rich-nguyen) [inline common/app/implicits/Requests.scala]: definitely a def!

**Pull request merged**